### PR TITLE
[SQL] Pytorch model inference runner in Spark SQL

### PR DESCRIPTION
**Pull request by eddyxu**
* Pytorch Runner that generates the pandas UDF to run model inference.
* Pre and post processing transforms for `fasterrcnn_resnet50_fpn`

### CODE REVIEW

File: python/rikai/torch/transforms/fasterrcnn_resnet50_fpn.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+#  Copyright 2021 Rikai Authors
```
Comment (changhiskhan) [inline python/rikai/torch/transforms/fasterrcnn_resnet50_fpn.py:1]: do we want this as part of the base library code that we'd want to keep compat for across torchvision versions etc?

File: python/tests/spark/sql/codegen/test_fs.py [hunk 2]
```diff
@@ -67,3 +106,29 @@ def test_validate_misformed_spec():
...
+            # https://cocodataset.org/#explore?id=433013
+            Row(
+                uri="http://farm4.staticflickr.com/3726/9457732891_87c6512b62_z.jpg"  # noqa
+            ),
+        ],
+    )
+    df.createOrReplaceTempView("df")
+
+    predictions = spark.sql(
+        "SELECT ML_PREDICT(resnet_m, uri) as predictions FROM df"
```
[the remark this answers is not in the document]
Comment (eddyxu) [reply python/tests/spark/sql/codegen/test_fs.py:129]: This requires #90 Spark UDT support to be done in upstream spark. I will add some comments here to indicate that later we will replace them into UDT.

File: python/rikai/spark/sql/codegen/fs.py [hunk 3]
```diff
@@ -46,22 +47,44 @@
...
-    """Model Spec"""
+    """Model Spec.
+
+    Parameters
+    ----------
+    spec : str, bytes, dict or file object
+        String content of the serialized spec, or a dict
+    options : Dict[str, Any], optional
+        Additionally options. If the same option exists in spec already,
+        it will be override.
```
[the remark this answers is not in the document]
Comment (eddyxu) [reply python/rikai/spark/sql/codegen/fs.py]: done

File: python/rikai/spark/sql/codegen/fs.py [hunk 4]
```diff
@@ -46,22 +47,44 @@
...
+    validate : bool
```
[the remark this answers is not in the document]
Comment (eddyxu) [reply python/rikai/spark/sql/codegen/fs.py]: done

File: python/tests/spark/sql/codegen/test_fs.py [hunk 5]
```diff
@@ -12,12 +12,51 @@
...
+import torchvision
+from pyspark.sql import Row, SparkSession
 
 from rikai.spark.sql.codegen.fs import ModelSpec
 from rikai.spark.sql.exceptions import SpecError
 
 
+@pytest.fixture(scope="module")
+def resnet_spec():
+    with tempfile.TemporaryDirectory() as tmp_dir:
```
[the remark this answers is not in the document]
Comment (eddyxu) [reply python/tests/spark/sql/codegen/test_fs.py:32]: Added a comment. `tmp_dir` or `tmp_path` can not be used with a `module` scoped fixture like this.

File: python/rikai/torch/transforms/fasterrcnn_resnet50_fpn.py (hunk 1 above)
Comment (eddyxu) [reply python/rikai/torch/transforms/fasterrcnn_resnet50_fpn.py:1]: you meant maintaining something like `rikai-contrib` package?

File: python/tests/spark/sql/codegen/test_fs.py [hunk 6]
```diff
@@ -67,3 +106,29 @@ def test_validate_misformed_spec():
...
+            ),
+        ],
+    )
+    df.createOrReplaceTempView("df")
+
+    predictions = spark.sql(
+        "SELECT ML_PREDICT(resnet_m, uri) as predictions FROM df"
+    )
+    predictions.show()
+    predictions.printSchema()
```
[the remark this answers is not in the document]
Comment (eddyxu) [reply python/tests/spark/sql/codegen/test_fs.py]: done

File: python/rikai/torch/transforms/fasterrcnn_resnet50_fpn.py (hunk 1 above)
Comment (eddyxu) [reply python/rikai/torch/transforms/fasterrcnn_resnet50_fpn.py]: moved these files to `rikai.contrib` , we can later release this package separately.

**Comment by eddyxu** [comment]
@changhiskhan take another look?

File: python/rikai/contrib/torch/transforms/utils.py [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+
+from rikai.types.vision import Image
+
+
+def uri_to_pil(uri):
```
Comment (changhiskhan) [inline python/rikai/contrib/torch/transforms/utils.py:19]: @sadhen FYI if you're wondering about use cases for the spark issue you're working on. This is a good one.

**Review by changhiskhan — APPROVED** [review]

**Pull request merged**